script.js

```diff
@@ -1,4 +1,159 @@
+Player 1 will press submit and 2 dice are rolled
+Player 1 to select dice order
+Player 2 will press submit and 2 dice are rolled
+Player 2 to select dice order
+To determine who the winner is */
+
+var playerOneChoice = "";
+var playerOneDiceRoll = "";
+var playerOneDiceOne = "";
+var playerOneDiceTwo = "";
+
+var playerTwoChoice = "";
+var playerTwoDiceRoll = "";
+var playerTwoDiceOne = "";
+var playerTwoDiceTwo = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // refresh all variables to play again
+  if (playerTwoChoice !== "") {
+    playerOneChoice = "";
+    playerOneDiceRoll = "";
+    playerOneDiceOne = "";
+    playerOneDiceTwo = "";
+
+    playerTwoChoice = "";
+    playerTwoDiceRoll = "";
+    playerTwoDiceOne = "";
+    playerTwoDiceTwo = "";
+  }
+
+  // player one has not rolled the dice
+  if (playerOneDiceRoll == "") {
+    playerOneDiceOne = rollDices();
+    playerOneDiceTwo = rollDices();
+    playerOneDiceRoll = "rolled";
+    return `Welcome Player 1! <br><br>
+    You rolled...<br>
+    # Dice One: ${playerOneDiceOne}<br>
+    # Dice Two: ${playerOneDiceTwo} <br><br>
+    Please type in either '1' or '2' to continue... <br>
+    # Option 1: ${playerOneDiceOne}${playerOneDiceTwo} <br>
+    # Option 2: ${playerOneDiceTwo}${playerOneDiceOne}`;
+  }
+
+  // if player one has rolled the dice, ask player one to choose arrangement
+
+  // if player one chooses option 1 (arrangement of dice 1+2)
+  if (playerOneDiceRoll == "rolled" && playerOneChoice == "" && input == "1") {
+    playerOneChoice = Number(
+      String(playerOneDiceOne) + String(playerOneDiceTwo)
+    );
+    return `Player 1, your choice is ${playerOneChoice}. <br><br>
+      Player 2, please roll the dices.`;
+  }
+
+  // if player one chooses option 2 (arrangement of dice 2+1)
+  if (playerOneDiceRoll == "rolled" && playerOneChoice == "" && input == "2") {
+    playerOneChoice = Number(
+      String(playerOneDiceTwo) + String(playerOneDiceOne)
+    );
+    return `Player 1, your choice is ${playerOneChoice}. <br><br>
+      Player 2, please roll the dices.`;
+  }
+
+  // if player one does not choose any of the options
+  if (playerOneChoice == "") {
+    return `Player 1, please type in either '1' or '2' to continue... <br><br>
+      # Option 1: ${playerOneDiceOne}${playerOneDiceTwo} <br>
+      # Option 2: ${playerOneDiceTwo}${playerOneDiceOne}`;
+  }
+
+  // if player two has not rolled the dice, let player two roll dice
+  if (playerTwoDiceRoll == "") {
+    playerTwoDiceOne = rollDices();
+    playerTwoDiceTwo = rollDices();
+    playerTwoDiceRoll = "rolled";
+    return `Welcome Player 2! <br><br>
+    You rolled...<br>
+    # Dice One: ${playerTwoDiceOne}<br>
+    # Dice Two: ${playerTwoDiceTwo} <br><br>
+    Please type in either '1' or '2' to continue... <br>
+    # Option 1: ${playerTwoDiceOne}${playerTwoDiceTwo} <br>
+    # Option 2: ${playerTwoDiceTwo}${playerTwoDiceOne}`;
+  }
```

Comment:
this part from line 74 onwards is fairly repetitive, consider 

- refactoring into a common function that allows both players to a) roll dice, b) decide dice order
- another function to decide winner after both players have decided dice order

hope you see the benefit of this - imagine if we dont have just 2 players but say, 10 players. repeating these code blocks 10 times will be a bit unwieldy and hard to maintain

Reply:
the reason why i emphasise the point of repetition so much is because in programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code.